### PR TITLE
BCC 29/sign in 

### PR DESCRIPTION
**Pull request by Thato-Tladi**

### CODE REVIEW

File: beancards/src/main/java/com/bbdgrads/beancards/Security/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class SecurityConfig {
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.csrf().disable()
+                .authorizeHttpRequests((authz) -> authz
+                        .requestMatchers("/player/signin").permitAll()
+                        .anyRequest().authenticated()
+                )
+                .oauth2Client(); // Use oauth2Client()
```
Comment (batandwa-bbd) [inline beancards/src/main/java/com/bbdgrads/beancards/Security/SecurityConfig.java]: What does this do?

File: beancards/src/main/java/com/bbdgrads/beancards/Services/AuthenticationService.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.bbdgrads.beancards.Entities.Player;
+import com.bbdgrads.beancards.Repositories.PlayerRepository;
+import com.bbdgrads.beancards.Repositories.PlayerContactRepository;
+
+public interface AuthenticationService {
+    String exchangeCodeForGithubToken(String code);
+    Player signInWithGithubToken(String githubAccessToken);
+
+
```
Comment (batandwa-bbd) [inline beancards/src/main/java/com/bbdgrads/beancards/Services/AuthenticationService.java:11]: Unnecessary empty lines

File: beancards/src/main/resources/application.properties [hunk 3]
```diff
@@ -2,3 +2,7 @@ spring.datasource.driver-class-name=org.h2.Driver
 spring.datasource.url=jdbc:h2:mem:db;DB_CLOSE_DELAY=-1
 spring.datasource.username=sa
 spring.datasource.password=sa
+github.client.id=d095913ff9b55112e726
+github.client.secret=f33c2fd6103c26f3b996f2a625d82ddd5d16e51e
+spring.security.oauth2.client.registration.github.client-id=d095913ff9b55112e726
+spring.security.oauth2.client.registration.github.client-secret=f33c2fd6103c26f3b996f2a625d82ddd5d16e51e
```
Comment (batandwa-bbd) [inline beancards/src/main/resources/application.properties:8]: Create another ticket moving these to environment variables. They should never be in source control.

File: cli/src/main/java/com/traderg/cli/Services/HttpListenerService.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                    exchange.sendResponseHeaders(400, 0);
+                }
+                exchange.close();
+
+                // Stop server after handling the request
+                server.stop(0);
+            });
+            server.start();
+
+            System.out.println("Server started. Waiting for GitHub callback...");
```
Comment (batandwa-bbd) [inline cli/src/main/java/com/traderg/cli/Services/HttpListenerService.java:30]: We not supposed to have these

File: cli/src/main/java/com/traderg/cli/Services/HttpListenerService.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+            // Wait for the code to be set
+            while (authorizationCode.get().isEmpty()) {
+                Thread.sleep(100); // waiting for a code
+            }
+
+            return authorizationCode.get();
+        } catch (Exception e) {
+            System.out.println("Server failed: " + e.getMessage());
```
Comment (batandwa-bbd) [inline cli/src/main/java/com/traderg/cli/Services/HttpListenerService.java:39]: Same here. Should not be here.

File: cli/src/main/java/com/traderg/cli/App.java [hunk 6]
```diff
@@ -1,13 +1,27 @@
...
-        System.out.println( "Hello World!" );
+import com.traderg.cli.Services.BackendService;
+import com.traderg.cli.Services.BrowserService;
+import com.traderg.cli.Services.HttpListenerService;
+
+
+public class App {
+    public static void main(String[] args) {
+        if (args.length > 0 && args[0].equalsIgnoreCase("login")) {
+            String authUrl = "https://github.com/login/oauth/authorize?client_id=d095913ff9b55112e726&redirect_uri=http://localhost:8888/signin&scope=user";
```
Comment (GerniVisser-bbd) [inline cli/src/main/java/com/traderg/cli/App.java:11]: Does this look for a certain client id ? Can anyone else also login ?

**Review by batandwa-bbd — APPROVED** [review]

**Review by KeeganOReilly — APPROVED** [review]
Happy days. LGTM.

**Pull request merged**